Adding Node::default_child_parent to control where scene children go

dmchurch opened the pull request:
This addresses https://github.com/godotengine/godot-proposals/issues/8184, allowing a PackedScene to specify where to put child nodes when Editable Children is *not* enabled.

* *Bugsquad edit, closes: https://github.com/godotengine/godot-proposals/issues/8184*
* also closes: https://github.com/godotengine/godot-proposals/issues/5475

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

dmchurch committed:
Add default_child_parent property to Node

This allows a PackedScene to specify where to put child nodes when
Editable Children is *not* enabled. This hooks into the editor in
two instances:

1. When creating a child node under an instantiated subscene root
   if the subscene root has `default_child_parent` set.
2. When reparenting an existing node to an instantiated subscene root
   if the subscene root has `default_child_parent` set.

It also hooks into the core engine in one instance:

1. When instantiating a PackedScene, if:
   - it contains an external reference to a second PackedScene, AND
   - the root node of the second PackedScene has `default_child_parent`
     set, AND
   - the containing PackedScene declares a node as a child of the
     second PackedScene's root node.

Also adds some unit tests for subscene instantiation, since that isn't
exercised yet.

Closes https://github.com/godotengine/godot-proposals/issues/8184
Closes https://github.com/godotengine/godot-proposals/issues/5475

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

AThousandShips commented on editor/icons/SetDefaultChildParent.svg:
```diff
@@ -0,0 +1 @@
+<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="M4 5a2 2 0 1 0 -2 0v1.5a2 2 0 1 0 2.75 2.75h1.5A2 2 0 0 0 7 10v1.5a2 2 0 1 0 2.75 2.75h1.5v-2h-1.5A2 2 0 0 0 9 11.5V10a2 2 0 1 0-2.75-2.75h-1.5A2 2 0 0 0 4 6.5V5z" fill="#e0e0e0"/><path d="M11.25 14.25a2 2 0 1 0 0 -2z" fill="#70bafa"/><path d="M 6 2.5a8 6 0 0 1 8 6h1l-2 2l-2 -2h1a6 4 0 0 0 -6 -4z" fill="#5 ...
```
```suggestion
<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="M4 5a2 2 0 1 0 -2 0v1.5a2 2 0 1 0 2.75 2.75h1.5A2 2 0 0 0 7 10v1.5a2 2 0 1 0 2.75 2.75h1.5v-2h-1.5A2 2 0 0 0 9 11.5V10a2 2 0 1 0-2.75-2.75h-1.5A2 2 0 0 0 4 6.5V5z" fill="#e0e0e0"/><path d="M11.25 14.25a2 2 0 1 0 0 -2z" fill="#70bafa"/><path d="M 6 2.5a8 6 0 0 1 8 6h1l-2 2l-2 -2h1a6 4 0 0 0 -6 -4z" fill="#5fff97"/></svg>
```
You have to close the `path` element

dmchurch replied:
ahaha whoops, this is what happens when I handcraft the XML instead of installing an editor, tyvm 😅

dmchurch committed:
Add indirect_child property to TreeItem

This draws the TreeItem with a broken line to represent discontinuity
in the relationship to its parent. The Scene Tree editor uses this
to represent node hierarchy relationships where the displayed child
node isn't the direct child of the node displayed above it.
